genCode.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from io import open
+
+import json
+import os
+import re
+import six
+import textwrap
+
+from jinja2 import Environment, FileSystemLoader
+
+GENERATED_STRING = "# This file is generated!  Do not edit!"
+# ROUTING_KEY_WHITELIST is also defined in test/base/__init__.py
+# if you change it, please update it there as well.
+ROUTING_KEY_WHITELIST = ("name", "multipleWords", "constant")
```

Comment:
The blacklist is looking good.  We're not using `required`, but other than longer files, it's harmless to include.  https://github.com/escapewindow/taskcluster-client.py/commit/8f8970277a91d451b53863fad52fa376301e2a85

(I'm going to be merging all of this into the codegen branch, and this PR, once I've dealt with all the review comments.)